crates/snp_validation/src/lib.rs

```diff
@@ -288,94 +310,129 @@ impl Snp {
 
         // Should only be version 2
         if report.body.version != 2 {
-            return Err(anyhow!("snp report is an unknown version"));
+            bail!("snp report is an unknown version");
         }
 
         // Check policy
         if !report.body.policy.flags.contains(PolicyFlags::Reserved) {
-            return Err(anyhow!("snp guest policy mandatory reserved flag not set"));
+            bail!("snp guest policy mandatory reserved flag not set");
         }
 
         if report.body.policy.flags.contains(PolicyFlags::MigrateMA) {
-            return Err(anyhow!("snp guest policy migration flag was set"));
+            bail!("snp guest policy migration flag was set");
+        }
+
+        if report.body.policy.abi_major > report.body.current_major {
+            bail!("snp policy has higher abi major than firmware");
+        } else if report.body.policy.abi_minor > report.body.current_minor {
+            bail!("snp policy has higher abi minor than firmware");
```

Comment:
Is there no `patch` version to check? If not, there should be a comment about it

Reply:
No patch available to check, comment added.